### ISSUE TITLE
Process Queue needs a delay

### ISSUE THREAD

**Issue by PaulWieland** [apearson/insteon-plm#36]
If you attempt to send multiple commands to devices rapidly, they begin to timeout requiring a restart of the docker container. Incidentally, I'm not quite sure why restarting the container unhangs the modem**.

Simply sending "on" commands to two devices simultaneously is enough to break it.

**EDIT: I believe the modem isn't actually hung, its the modem's process queue which is hung - which is why the restart of docker fixes it.

**Comment by PaulWieland** [comment]
I am experimenting with putting a 100ms delay around the `processQueue` `callback` and so far (with very limited testing) it seems to work more reliably. I believe this is working because it is forcing 100ms between messages sent to the modem.

```JavaScript
	private processQueue = async (task: QueueTaskData, callback: AsyncResultCallback<AnyPacket>) => {

		// Once we hear an echo (same command back) the modem is ready for another command
		this.once(['p', task[1].toString(16)], (d: Packet.Packet) => {
			if(d.ack){
				const packet = d as Packet.SendInsteonMessage;

				if(packet.ack){
					setTimeout(function(){
						callback(null, packet);
					}, 100);
				}
				else{
					callback(Error('Could not complete task'));
				}
			}
			else{
				setTimeout(function(){
					callback(null, d);
				}, 100);
			}
		});
		// Attempting to write command to modem
		try{
			const isSuccessful = this.port.write(task);

			if(!isSuccessful)
				callback(Error('Could not write to modem'));
		}
		catch(error){
			callback(error);
		}
	}
```

**Comment by PaulWieland** [comment]
Rapidly sending commands causes a timeout error after a certain point. I suspect the bluebird `timeout()` starts ticking when the item is added to the queue, and not when it starts processing.

This scenario can be reproduced by rapidly clicking an inject node to send the same `command` (dim up one step for example) to the PLM node repeatedly. Note that the modem isn't freezing, so the setTimeout delay is having the desired effect in that regard.

![image](https://user-images.githubusercontent.com/4663918/71481512-1a85a800-27cc-11ea-8b75-7dfaabe3f00f.png)

**Comment by PaulWieland** [comment]
Update: I removed all bluebird `timeout()` calls and changed the `processQueue` to the below. The logic for what to do when a timeout occurs needs work. Further, i need to come up with a reliable condition for triggering a timeout to do further testing.

With this `processQueue` logic, I can flood the PLM with as many tasks as I want and it doesn't crash.

```JavaScript
private processQueue = async (task: QueueTaskData, callback: AsyncResultCallback<AnyPacket>) => {
	let timer:NodeJS.Timer;
	
	// Once we hear an echo (same command back) the modem is ready for another command
	this.once(['p', task[1].toString(16)], (d: Packet.Packet) => {
		
		debug(`pong`);
		clearTimeout(timer);

		if(d.ack){
			const packet = d as Packet.SendInsteonMessage;

			if(packet.ack){
				debug('waiting for ack setTimeout');
				setTimeout(function(){
					debug('ack setTimeout finished');
					callback(null, packet);
				}, 350);
			}
			else{
				callback(Error('Could not complete task'));
			}
		}
		else{
			debug('waiting for setTimeout')
			setTimeout(function(){
				debug('setTimeout finished')
				callback(null, d);
			}, 350);
		}
	});

	// Attempting to write command to modem
	try{
		timer = setTimeout(function(){
			debug("No response received within 2 seconds");
			// callback(Error('No response received within 2 seconds'));
		},2000);
		
		const isSuccessful = this.port.write(task);
		
		debug(`ping`);

		if(!isSuccessful)
			callback(Error('Could not write to modem'));
	}
	catch(error){
		callback(error);
	}
}

```

**Comment by PaulWieland** [comment]
I found in testing that the separate device queues are causing problems.

With the above PLM queue logic and a setTimeout of 450ms ensuring that the modem is processing every command and response in a serialized fashion, there is no need for separate async queues for each device. 

To test, I simplified the device's `processQueue` to:

```JavaScript
	private processQueue = async (task: DeviceCommandTask, callback: AsyncResultCallback<Packet.Packet>) => {
		const isSuccessful = !!task.extendedData ? await this.modem.sendExtendedCommand(this.address, task.cmd1, task.cmd2, task.extendedData, task.flags)
		                                         : await this.modem.sendStandardCommand(this.address, task.cmd1, task.cmd2, task.flags);
												 
		callback(null);
		
	}
```

But in fact the device queue could and should be removed all together because it's redundant with the plm queue.

For testing, I took two dimmers, one at full bright and one off. I then sent 32 `dim up one step` commands to the dimmer which was off and 32 `dim down one step` commands to the one which was on. All 64 commands were sent simultaneously and the two dimmers step up/down without missing any messages. The process is slow due to the 450ms setTimeout in the PLM's `processQueue`, but I think that's the limitation of Insteon.

![image](https://user-images.githubusercontent.com/4663918/71523229-1faf2980-2896-11ea-888e-189ba32c07e4.png)
